test/schedulers.py

```diff
@@ -0,0 +1,97 @@
+import test_utils
+from torch.optim.lr_scheduler import _LRScheduler
+
+MIN_DELTA_TO_UPDATE_LR = 0.000001
+NUM_WARMUP_EPOCHS = 1
+
+
+class WarmupAndExponentialDecayScheduler(_LRScheduler):
+  """Update the learning rate of wrapped optimizer based on epoch and step.
+  
+  Args:
+    optimizer: Instance of torch.optim.Optimizer. Learning rate will be changed.
+    num_steps_per_epoch: int, the number of steps required to finish 1 epoch.
+    divide_every_n_epochs: After this number of epochs, learning rate will be
+        divided by the `divisor` param.
+    divisor: The learning rate will be divided by this amount when
+        epoch % divide_every_n_epochs == 0 (epoch 0 is excluded).
+    summary_writer: Instance of `torch.utils.tensorboard.SummaryWriter`. If
+        provided, learning rate will be logged during calls to step if step
+        is called with write_to_summary=True. If summary_writer is None, then
+        no logging happens.
+  """
+  def __init__(self, optimizer, num_steps_per_epoch, divide_every_n_epochs=20,
+               divisor=5, summary_writer=None):
+    self.num_steps_per_epoch = num_steps_per_epoch
```

Comment:
Use `_` prefix for members. unless they are public. Which they should not AFIACT.

Reply:
Switched to prefix. None of these should be public